### PR TITLE
[path_parsing] Initial import

### PR DESCRIPTION
**Pull request by stuartmorgan**
Imports https://github.com/dnfield/dart_path_parsing into this repository, with history, and updates it to follow repository conventions:
- Updates minimum SDK support to match our minimum
- Switches to repository analysis options, fixing or suppressing all resulting warnings
- Standardizes pubspec.yaml to follow our style/conventions
- Removes the use of pubspec.lock, per repo conventions

### CODE REVIEW

**Comment by stuartmorgan** [comment]
@zanderso Would you like to edit the README to add a note?

Also, currently we use CODEOWNERS in this repo; who should I list for this and the follow-up packages?

**Comment by zanderso** [comment]
Yes, I'll edit the README. Working on the text, but I think this PR doesn't need to block on it?

A while back, @domesticmouse volunteered to be the owner for these for the purposes of mechanical updates with more involved issues in `flutter_svg` being owned by @jonahwilliams and the engine team.

File: third_party/packages/path_parsing/README.md
Comment (domesticmouse) [inline third_party/packages/path_parsing/README.md:5]: Nit: Add trailing EOL at EOF

**Review by domesticmouse — APPROVED** [review]

**Pull request merged**
